src/lib/rule2A.ts

```diff
@@ -0,0 +1,56 @@
+import {Context} from './context';
+
+/**
+ * Looks at a variety of characteristics (CSS, size on screen, attributes)
+ * to determine if 'node' should be considered hidden
+ * @param node - node whose hidden-ness is being calculated
+ * @return - whether or not the node is considered hidden
+ */
+function isHidden(node: Node): boolean {
+  if (node instanceof HTMLElement) {
+    const nodeStyle = window.getComputedStyle(node);
+
+    const hiddenByStyle = (
+      nodeStyle.visibility === 'hidden' ||
+      // measurement of pixels used on screen (checks for display:none)
+      node.offsetHeight <= 0 ||
+      node.offsetWidth <= 0
+    );
+
+    const hiddenByAttribute = (
+      node.closest('[hidden],[aria-hidden="true"]') !== null
+    );
+
+    return (hiddenByAttribute || hiddenByStyle);
```

Comment:
style nit: you are potentially computing unnecessary things here by not returning early. 

Returning early can also help with readability & avoid levels of nesting.

I'd suggest organizing the code like the following:

```typescript
function isHidden(node: Node): boolean {
  if (!(node instanceof HTMLElement)) {
    return false;
  }

  const visibility = window.getComputedStyle(node).visibility;
  const notDisplayed = node.offsetHeight === 0 && node.offsetWidth === 0;
  if (visibility === 'hidden' || notDisplayed) {
    return true;
  }

  const hiddenAncestor = node.closest('[hidden],[aria-hidden="true"]');
  if (hiddenAncestor !== null) {
    return true;
  }

  return false;
}
```